app/components/polls/questions/question_component.rb

```diff
@@ -48,28 +48,41 @@ def multiple_choice_help_text
       )
     end
 
-    def multiple_choice_field(option)
-      choice_field(option) do
+    def choice_field(option)
+      label_tag("web_vote_option_#{option.id}") do
+        html = input_tag(option) + option.title
+        html += open_text_tag(option) if option.open_text?
+        html
+      end
+    end
+
+    def input_tag(option)
+      if multiple_choice?
         check_box_tag "web_vote[#{question.id}][option_id][]",
                       option.id,
                       checked?(option),
                       id: "web_vote_option_#{option.id}"
-      end
-    end
-
-    def single_choice_field(option)
-      choice_field(option) do
+      else
         radio_button_tag "web_vote[#{question.id}][option_id]",
                          option.id,
                          checked?(option),
                          id: "web_vote_option_#{option.id}"
       end
     end
 
-    def choice_field(option, &block)
-      label_tag("web_vote_option_#{option.id}") do
-        block.call + option.title
-      end
+    def open_text_tag(option)
+      text_field_tag(
+        "web_vote[#{question.id}][answer][#{option.id}]",
+        existing_text_for(option),
+        id: "web_vote_option_#{option.id}_text",
+        disabled: disabled?,
+        class: "open-text",
+        data: { selects: "web_vote_option_#{option.id}" }
+      )
+    end
+
+    def existing_text_for(option)
+      question.answers.find_by(author: current_user, option: option)&.answer.to_s
```

Comment:
Right now the answer is lost when submitting an invalid form, so we probably have to use `form.object.answers` as we do in `existing_answer` :thinking:.

By the way, is there a difference between `existing_text_for` and `existing_answer`?